subprojects/enterprise/src/integTest/groovy/org/gradle/internal/enterprise/GradleEnterprisePluginCheckInIntegrationTest.groovy

```diff
@@ -121,8 +124,36 @@ class GradleEnterprisePluginCheckInIntegrationTest extends AbstractIntegrationSp
         getMinimumPluginVersionForConfigurationCaching() | true
     }
 
+    @Requires(IntegTestPreconditions.NotConfigCached)
+    def "shows warning message when Gradle Enterprise plugin version is used with isolated projects enabled"() {
+        given:
+        plugin.runtimeVersion = plugin.artifactVersion = pluginVersion
+        applyPlugin()
+        settingsFile << """
+            println "present: " + services.get($GradleEnterprisePluginManager.name).present
+        """
+
+        when:
+        succeeds("t", "-D${StartParameterBuildOptions.IsolatedProjectsOption.PROPERTY_NAME}=true")
```

Comment:
❓Should we avoid referencing the property name here, so that when we change it, one can see what tests/use-cases become broken?

Reply:
🤔 Hm, sorry but can't get it how we could benefit from broken tests after changing property name. At opposite - with referencing the property name, test becoming resistant to property name change, what is reasonable IMO.

Or I missing something?

Reply:
It may not matter as much in this particular case, but I think it's a good practice to avoid as much as possible to refer to production code in the integration tests. You can keep the reference instead of inlining the parameter name, but I still want to explain my reasoning.

> test becoming resistant to property name change, what is reasonable

I can agree with this statement in respect to unit tests, where the goal is to check that some code-under-test is working correctly. Even if you make some changes to the code, the test validates an "invariant" that effectively ignores the value of the property name, but still "does the expected thing".

The integration tests serve an additional purpose of preserving a user contract. So if the behavior of the code changes (and a parameter name change is a behavior change), the tests must catch it and fail. This makes sure that changes are not made accidentally during refactorings or however else. It additionally forces behavior changes to be accompanied by explicit integration test changes. This ties the "behavior contract" together with production code.

So yes, I think an integration test should absolutely break if we change a parameter name. And any breaking change must fail some integration tests, otherwise we should improve coverage.

Again, it's a slight exaggeration here, but you can see that the majority of the Gradle's integration tests follow this practice.

Reply:
Interesting, thank you!